src/project-manager.ts

```diff
@@ -316,12 +315,8 @@ export class ProjectManager implements Disposable {
 		if (observable) {
 			return observable;
 		}
-		const parts = url.parse(uri);
-		if (!parts.pathname) {
-			return Observable.throw(new Error(`Invalid URI ${uri}`));
-		}
 		// TypeScript works with file paths, not URIs
-		const filePath = parts.pathname.split('/').map(decodeURIComponent).join('/');
+		const filePath = util.uri2path(uri);
```

Comment:
I don't see anything incorrect about the now deleted URI-to-path conversion. What is the difference here, except that the version before handled non-file URIs and query, hash components? Does the TS service require paths to be in the format of the host OS?